src/classes/ACCT_IndividualAccounts_TDTM.cls

```diff
@@ -52,567 +51,718 @@ public class ACCT_IndividualAccounts_TDTM extends TDTM_Runnable {
+    /*******************************************************************************************************
+    * @description Calls correct methods to process after insert and after update
+    * logic and sets recursion flag.
+    * @param contacts a list of Contacts from trigger new.
+    * @param oldContacts a list of Contacts from trigger old.
+    * @param triggerAction which trigger event (BeforeInsert, AfterInsert, etc.).
+    * @return void.
+    */
+    private void handlesAfterInsertUpdate(Contact [] newContacts, Contact [] oldContacts, TDTM_Runnable.Action triggerAction) {
+        List<Contact> contactsNeedAccounts = new List<Contact>();
+        HandleUpdateWrapperLogic updateWrapper;
+        HandleInsertWrapperLogic insertWrapper;
 
-            strSoql += 'Phone, Fax from Contact where Id IN :contacts';
-            contacts = database.query(strSoql);
-        }
+        Integer i = 0;
+        for (Contact c : newContacts) {
 
-        // AFTER DELETE
-        if (triggerAction == TDTM_Runnable.Action.AfterDelete && !TDTM_ProcessControl.getRecursionFlag(TDTM_ProcessControl.registeredTrigger.ACCT_IndividualAccounts_TDTM_After_Delete)) {
-            Set<Id> relatedAccountIds = new Set<Id>();
-            List<Account> relatedAccounts = new List<Account>();
-            for (Contact c : oldContacts) {
-                listContactAccountDelete.add(c);
-                //Collect related account ids
-                relatedAccountIds.add(c.AccountId);
+            //Account needs to be created if (1) Account does not exist or (2) Account model is not blank
+            if (c.AccountId == NULL && defaultRecTypeId != NULL) {
+                contactsNeedAccounts.add(c);
             }
 
-            //Re-query to get correct Account values
-            relatedAccounts = [SELECT Id,
-                                    RecordTypeId
-                                FROM Account
-                                WHERE Id IN :relatedAccountIds];
-            for (Account account : relatedAccounts) {
-                // Rename the account if a contact is deleted from a household account
-                if (checkHouseholdNaming(account.RecordTypeId)) {
-                    setAccountIdHHToRename.add(account.Id);
+            //AFTER INSERT
+            if (triggerAction == TDTM_Runnable.Action.AfterInsert &&
+                !TDTM_ProcessControl.getRecursionFlag(TDTM_ProcessControl.registeredTrigger.ACCT_IndividualAccounts_TDTM_After_Insert)) {
+                    insertWrapper = new HandleInsertWrapperLogic (c);
+                }
+
+            //AFTER UPDATE
+            if (triggerAction == TDTM_Runnable.Action.AfterUpdate &&
+                !TDTM_ProcessControl.getRecursionFlag(TDTM_ProcessControl.registeredTrigger.ACCT_IndividualAccounts_TDTM_After_Update)) {
+                    updateWrapper = new HandleUpdateWrapperLogic (c, oldContacts, triggerAction, i);
```

Comment:
Same comment/nit as Ryan's above - can it be `new HandleUpdateWrapperLogic(c, ...);` ?